record.c

```diff
@@ -248,8 +248,16 @@ int janus_recorder_close(janus_recorder *recorder) {
 		/* We need to rename the file, to remove the temporary extension */
 		char newname[1024];
 		memset(newname, 0, 1024);
-		g_snprintf(newname, strlen(recorder->filename)-strlen(rec_tempext), "%s", recorder->filename);
-		if(rename(recorder->filename, newname) != 0) {
+		char tmpname[1024];
+		memset(tmpname, 0, 1024);
+		if(recorder->dir) {
+			g_snprintf(newname, strlen(recorder->filename)-strlen(rec_tempext), "%s/%s", recorder->dir, recorder->filename);
```

Comment:
We should actually check if `strlen(recorder->filename)-strlen(rec_tempext)` (or `strlen(recorder->filename)` when the temporary extension is not used) is higher than `sizeof(newname)`, as otherwise any of those `g_snprintf` may overflow. It's unlikely but adding the check won't hurt. This is actually my fault as it was part of the original patch, but I guess fixing it here makes sense.